ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
Agreed. I think the confusion here is over all the bit logic in the proposed code. What I had proposed is that we take the 2^15th bit (the MSB) of the local jobid as the sentinel. Thus, all you have to do is check the sign of the ORTE_LOCAL_JOBID. If it is positive, then it is the sentinel isn't set - if it is negative, then the sentinel is set. There is no reason to be bit-shifting things.

Likewise, when you create the sentinel, you create the process name with the negative of the the local jobid. Then we adjust the orte_create_jobid code so it only uses values up to 2^15 - 1.

Does that help?